automation/compare-gems

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env ruby
+require 'bundler'
+require 'json'
+require 'open3'
+
+EXCLUDED_GROUPS = [:test, :development]
+SCL_PREFIXES = JSON.load(File.read(File.join(__dir__, '..', 'scl_prefixes.json')))
+
+def package_name(gem_name)
+  # TODO: EL8 has no SCL
+  "tfm-rubygem-#{gem_name}"
+end
+
+def get_upstream_versions(filename)
+  Bundler.settings.set_local(:without,  EXCLUDED_GROUPS)
+  Bundler.settings.set_local(:with, [])
+
+  definition = Bundler::Dsl.evaluate(filename, nil, true)
+  definition.resolve_remotely!
+
+  Hash[definition.requested_specs.map do |spec|
+    unless SCL_PREFIXES.include?(spec.name)
+      [package_name(spec.name), spec.version.to_s]
+    end
+  end.compact]
+end
+
+def get_packaging_versions
+  stdout, _, status = Open3.capture3(File.join(__dir__, 'build_cache'))
+  raise Exception('Failed to build cache') unless status.success?
+  JSON.load(stdout)
+end
+
+def main
+  filename = ARGV.shift || 'Gemfile'
+
+  upstream_versions = get_upstream_versions(filename)
+  packaging_versions = get_packaging_versions
+
+  upstream_versions.each do |package, upstream_version|
+    packaging = packaging_versions[package]
+    if packaging.nil?
+      puts "#{package} is unpackaged but may be provided by an SCL - should be #{upstream_version}"
```

Comment:
how expensive would it be to look into `rails-packaging` to at least look at our own scl?

Reply:
the following would be found in our scl:

```
tfm-rubygem-mimemagic is unpackaged but may be provided by an SCL - should be 0.3.3
tfm-rubygem-marcel is unpackaged but may be provided by an SCL - should be 0.3.3
tfm-rubygem-activestorage is unpackaged but may be provided by an SCL - should be 5.2.1
```

Reply:
Not sure. However, if there's a mismatch here then the scl_prefixes file should be synced because it would also generate incorrect dependencies.

Reply:
Marking as resolved: we no longer have rails-packaging and everything is in our repository. Sometimes issues fix themselves over time ;)